azurelinuxagent/ga/exthandlers.py

```diff
@@ -1001,6 +1001,7 @@ def report_ext_handler_status(self, vm_status, ext_handler, goal_state_changed):
                     heartbeat = ext_handler_i.collect_heartbeat()
                     if heartbeat is not None:
                         handler_status.status = heartbeat.get('status')
+                        handler_status.message = parse_formatted_message(heartbeat.get('formattedMessage'))
```

Comment:
you should check that the heartbeat actually has a 'formattedMessage')

Reply:
why do we need extra check. The get call would handle if it's not available. It will return None if property does not present and same applies to status field.

Reply:
@nagworld9 we need the check so that we do not assign None to these properties

Reply:
I see.  At this point we are overriding the handler status file content with heartbeat not with default None values. So makes sense to check here.